test: use unique tmpdirs for each test (#28858)
nodejs/node: test/common/README.md

Deletes and recreates the testing temporary directory.

> The first time refresh is run it adds a listener to process `'exit'` that

Reviewer: Optional nit:

```suggestion
The first time `refresh()` runs,  it adds a listener to process `'exit'` that
```

cleans the temporary directory. Thus, every file under `tmpdir.path` needs to
be closed before the test completes. A good way to do this is to add a

> listener to process `'beforeExit'`. If a file needs to be left open until Node

Reviewer: Optional nit:

```suggestion
listener to process `'beforeExit'`. If a file needs to be left open until Node.js
```

or

```suggestion
listener to process `'beforeExit'`. If a file needs to be left open until `node`
```

> completes, use a child process and call `refresh` only in the parent.

Reviewer: Optional nit:

```suggestion
completes, use a child process and call `refresh()` only in the parent.
```
